File: user-service/src/main/java/socialapp/userservice/model/entity/Relation.java
package socialapp.userservice.model.entity;

import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import lombok.Setter;
import socialapp.userservice.model.enums.RelationStatus;

@MappedSuperclass
@Getter
@Setter
public abstract class Relation extends BaseEntity{
    @ManyToOne
    @JoinColumn
    private User receiver;
    @ManyToOne
    @JoinColumn
    private User sender;

    public abstract RelationStatus getRelationStatus();
}

File: kt-user-service/src/main/kotlin/socialapp/ktuserservice/model/entity/Relation.kt
package socialapp.ktuserservice.model.entity
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.MappedSuperclass

@MappedSuperclass
abstract class Relation : BaseEntity() {

    @ManyToOne
    @JoinColumn
    var receiver: User? = null

    @ManyToOne
    @JoinColumn
    var sender: User? = null

}